staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -487,13 +489,17 @@ func (f *featureGate) OverrideDefaultAtVersion(name Feature, override bool, ver
 // GetAll returns a copy of the map of known feature names to feature specs for the current emulationVersion.
 func (f *featureGate) GetAll() map[Feature]FeatureSpec {
 	retval := map[Feature]FeatureSpec{}
-	for k, v := range f.GetAllVersioned() {
-		spec := f.featureSpecAtEmulationVersion(v)
+	f.lock.Lock()
+	versionedSpecs := f.GetAllVersioned()
+	emuVer := f.EmulationVersion()
```

Comment:
sync.Mutex is not re-entrant: https://go.dev/play/p/2mXVxTlzqHB

This should get caught be tests.. If it doesn't lets make sure to improve the testing.

Reply:
My mistake, I was looking at `SetEmulationVersion` and observing that it also calls lock, but this calls `EmulationVersion`, which is safe.